exec.ts

```diff
@@ -0,0 +1,40 @@
+import stream = require('stream');
+
+import { WebSocketHandler } from './web-socket-handler';
+import { KubeConfig } from '@kubernetes/typescript-node';
+
+export class Exec {
+    'handler': WebSocketHandler;
+
+    public constructor(config: KubeConfig) {
+        this.handler = new WebSocketHandler(config);
+    }        
+
+    // TODO: make command an array and support multiple args
+    public exec(namespace: string, podName: string, containerName: string, command: string, stdout: stream.Writable | any, stderr: stream.Writable | any) {
+        var hasStdout = stdout != null;
+        var hasStderr = stderr != null;
+        var path = `/api/v1/namespaces/${namespace}/pods/${podName}/exec?command=${command}&container=${containerName}&container=nginx&stderr=${hasStderr}&stdout=${hasStdout}`;        
+        this.handler.connect(path, null, (stream: number, buff: Buffer) => {
+            if (buff.length < 1) {
+                return;
+            }
+            if (stream == 1) {
+                stdout.write(buff);
+            } else if (stream == 2) {
+                stderr.write(buff);
+            } else if (stream == 3) {
+                // stream closing.
+                // TODO Parse status here.
+                if (stdout) {
+                    stdout.end();
+                }
+                if (stderr) {
+                    stderr.end();
+                }
+            } else {
+                console.log("Unknown stream: " + stream);
+            }
+        });
+    }
+}
```

Comment:
can you add a newline at the end of the file?

Reply:
done.